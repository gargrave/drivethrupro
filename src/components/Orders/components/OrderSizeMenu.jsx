import React from 'react'
import { array, func, number } from 'prop-types'

const OrderSizeMenu = props => {
  return (
    <span className="input-group">
      <label htmlFor="size">Size: </label>
      <select name="size" id="size" value={props.sizeIdx} onChange={props.handleSizeChange}>
        {props.availableSizes.map((size, idx) => {
          return (
            <option key={idx} value={idx}>
              {size.name}
            </option>
          )
        })}
      </select>
    </span>
  )
}

OrderSizeMenu.propTypes = {
  sizeIdx: number.isRequired,
  availableSizes: array.isRequired,
  handleSizeChange: func.isRequired
}

export default OrderSizeMenu
