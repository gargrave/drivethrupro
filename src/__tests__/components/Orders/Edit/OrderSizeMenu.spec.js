import React from 'react'
import { shallow } from 'enzyme'

import OrderSizeMenu from '../../../../components/Orders/Edit/OrderSizeMenu'

const sizes = [{ name: 'Small', price: 49 }, { name: 'Medium', price: 69 }, { name: 'Large', price: 89 }]

describe('OrderSizeMenu', () => {
  let component
  let props

  beforeEach(() => {
    props = {
      sizeIdx: 1,
      availableSizes: sizes,
      handleSizeChange: jest.fn()
    }
    component = shallow(<OrderSizeMenu {...props} />)
  })

  test('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  test('pre-selects the option passed into props', () => {
    const select = component.find('.order-size-menu')
    const expected = props.sizeIdx
    const actual = select.props().value
    expect(actual).toEqual(expected)
  })

  test('renders the correct number of options', () => {
    const actual = component.find('.order-size-menu option').length
    const expected = sizes.length
    expect(actual).toEqual(expected)
  })

  test('calls the "change" callback when an option is selected', () => {
    expect(props.handleSizeChange.mock.calls.length).toEqual(0)
    component.find('.order-size-menu').simulate('change')
    expect(props.handleSizeChange.mock.calls.length).toEqual(1)
  })
})
