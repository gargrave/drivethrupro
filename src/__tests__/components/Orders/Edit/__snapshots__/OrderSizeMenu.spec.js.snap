// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSizeMenu renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="input-group"
>
    <label
        htmlFor="size"
    >
        Size: 
    </label>
    <select
        className="order-size-menu"
        id="size"
        name="size"
        onChange={[Function]}
        value={1}
    >
        <option
            value={0}
        >
            Small
        </option>
        <option
            value={1}
        >
            Medium
        </option>
        <option
            value={2}
        >
            Large
        </option>
    </select>
</span>,
  "nodes": Array [
    <span
      className="input-group"
>
      <label
            htmlFor="size"
      >
            Size: 
      </label>
      <select
            className="order-size-menu"
            id="size"
            name="size"
            onChange={[Function]}
            value={1}
      >
            <option
                  value={0}
            >
                  Small
            </option>
            <option
                  value={1}
            >
                  Medium
            </option>
            <option
                  value={2}
            >
                  Large
            </option>
      </select>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <OrderSizeMenu
        availableSizes={
                Array [
                        Object {
                          "name": "Small",
                          "price": 49,
                        },
                        Object {
                          "name": "Medium",
                          "price": 69,
                        },
                        Object {
                          "name": "Large",
                          "price": 89,
                        },
                      ]
        }
        handleSizeChange={[Function]}
        sizeIdx={1}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "availableSizes": Array [
            Object {
              "name": "Small",
              "price": 49,
            },
            Object {
              "name": "Medium",
              "price": 69,
            },
            Object {
              "name": "Large",
              "price": 89,
            },
          ],
          "handleSizeChange": [Function],
          "sizeIdx": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="input-group"
>
          <label
                    htmlFor="size"
          >
                    Size: 
          </label>
          <select
                    className="order-size-menu"
                    id="size"
                    name="size"
                    onChange={[Function]}
                    value={1}
          >
                    <option
                              value={0}
                    >
                              Small
                    </option>
                    <option
                              value={1}
                    >
                              Medium
                    </option>
                    <option
                              value={2}
                    >
                              Large
                    </option>
          </select>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="input-group"
>
          <label
                    htmlFor="size"
          >
                    Size: 
          </label>
          <select
                    className="order-size-menu"
                    id="size"
                    name="size"
                    onChange={[Function]}
                    value={1}
          >
                    <option
                              value={0}
                    >
                              Small
                    </option>
                    <option
                              value={1}
                    >
                              Medium
                    </option>
                    <option
                              value={2}
                    >
                              Large
                    </option>
          </select>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OrderSizeMenu
    availableSizes={
        Array [
            Object {
              "name": "Small",
              "price": 49,
            },
            Object {
              "name": "Medium",
              "price": 69,
            },
            Object {
              "name": "Large",
              "price": 89,
            },
          ]
    }
    handleSizeChange={[Function]}
    sizeIdx={1}
/>,
}
`;
